### PR TITLE
feat(eslint-config): use `eslint>=7` as peerDependency

### PR DESCRIPTION
**Pull request by splincode**
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

Users can install manually eslint 8 as optional package instead of transitive dependency

### CODE REVIEW

File: packages/eslint-config/package.json [hunk 1]
```diff
@@ -23,7 +23,6 @@
     "@babel/eslint-plugin": "^7.19.1",
     "@typescript-eslint/eslint-plugin": "^5.45.1",
     "@typescript-eslint/parser": "^5.45.1",
-    "eslint": "^7.5.0",
```
Comment (alexkvak) [inline packages/eslint-config/package.json:26]: you should add to README.md that implicit eslint installation is needed

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
 
     - uses: actions/setup-node@v2
       with:
-        node-version: '14.x'
+        node-version: 18.x
```
Comment (alexkvak) [inline .github/workflows/release.yml:27]: what is the point here?

File: packages/eslint-config-angular/package.json [hunk 3]
```diff
@@ -37,7 +37,13 @@
...
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-sort-class-members": "^1.14.1",
-    "eslint-plugin-unicorn": "^44.0.2"
+    "eslint-plugin-unicorn": "^44.0.2",
+    "eslint-plugin-decorator-position": "^5.0.1",
+    "eslint-plugin-file-progress": "^1.3.0",
+    "eslint-plugin-html": "^7.1.0",
+    "eslint-plugin-rxjs": "^5.0.2",
+    "eslint-plugin-rxjs-angular": "^2.0.0",
+    "eslint-plugin-simple-import-sort": "^7.0.0"
```
Comment (alexkvak) [inline packages/eslint-config-angular/package.json:46]: all these new packages are used anywhere?

File: .npmrc [hunk 4]
```diff
@@ -1,2 +1,6 @@
+lockfileVersion=2
 scripts-prepend-node-path=true
 legacy-peer-deps=true
+engine-strict=false
+ignore-engines=true
+loglevel=error
```
Comment (alexkvak) [inline .npmrc:6]: looks not nice

File: .github/workflows/release.yml (hunk 2 above)
Comment (splincode) [reply .github/workflows/release.yml:27]: what's the difference from which `node.js` to publish in npm?

File: .npmrc (hunk 4 above)
Comment (splincode) [reply .npmrc:6]: use from here:
https://github.com/Tinkoff/taiga-ui/blob/main/.npmrc#L3

File: .github/workflows/release.yml (hunk 2 above)
Comment (alexkvak) [reply .github/workflows/release.yml:27]: exactly :)

File: packages/eslint-config-angular/package.json (hunk 3 above)
Comment (splincode) [reply packages/eslint-config-angular/package.json:46]: image

**Pull request closed without merging**